File: C#/TemplateMethod/WordCVReportGeneration.cs
using dotnet.TemplateMethod.Models;

namespace dotnet.TemplateMethod;

public class WordCVReportGeneration : CvReportGeneration
{
    protected override Data ExtractData(Uri cvPath)
    {
        Console.WriteLine("Extracting data from word...");
        
        return new Data();
    }
}

File: typescript/patterns/template/word-cv-report-generation.ts
import { CVReportGeneration } from "./cv-report-generation";
import { Data } from "./data";
import { FileInfo } from "./file-info";

export class WordCVReportGeneration extends CVReportGeneration {
  override extractData(file: FileInfo): Data {
    console.log("extracting data from word file");
    return new Data();
  }
}
